test/axes/baseAxisTests.ts

```diff
@@ -206,4 +206,661 @@ describe("BaseAxis", () => {
+
+          svg.remove();
+        });
+
+        it("re-renders when annotated ticks are set after render", () => {
+          let annotatedTicks = [new Date(1994, 11, 17), new Date(1995, 11, 17)];
+          let scale = new Plottable.Scales.Time();
+          scale.domain([new Date(1994, 11, 17), new Date(1995, 11, 17)]);
+          let axis = new Plottable.Axis(scale, "bottom");
+          axis.annotationsEnabled(true);
+          let svg = TestMethods.generateSVG(300, 300);
+          axis.renderTo(svg);
+
+          assert.strictEqual(axis.content().selectAll(".annotation-label").size(), 0, "no annotations by default");
+
+          axis.annotatedTicks(annotatedTicks);
+
+          assert.strictEqual(axis.content().selectAll(".annotation-label").size(), annotatedTicks.length, "annotated ticks annotated");
+
+          let annotationFormatter = axis.annotationFormatter();
+          let annotationLabels = axis.content().selectAll(".annotation-label");
+          annotationLabels.each(function(d, i) {
+            let d3this = d3.select(this);
+            assert.strictEqual(d3this.text(), annotationFormatter(annotatedTicks[i]), `annotated tick ${i} has been formatted`);
+          });
+          svg.remove();
+        });
+      });
+
+      describe("element placement in relation to annotation rectangle", () => {
+
+        let scale: Plottable.Scales.ModifiedLog;
+        let annotatedTicks: number[];
+
+        beforeEach(() => {
+          scale = new Plottable.Scales.ModifiedLog();
+          annotatedTicks = [3, 100, 250];
+          scale.domain([0, 300]);
+        });
+
+        it("renders annotations at the axis baseline extending downwards with a bottom orientation axis", () => {
+          let axis = new Plottable.Axis(scale, "bottom");
+          axis.annotatedTicks(annotatedTicks);
+          axis.annotationsEnabled(true);
+
+          let svg = TestMethods.generateSVG(300, 300);
+          axis.renderTo(svg);
+
+          let annotationRects = axis.content().selectAll(".annotation-rect");
+
+          let annotationCircles = axis.content().selectAll(".annotation-circle");
+
+          assert.strictEqual(annotationCircles.size(), annotationRects.size(), "same number of annotation circles as rects");
+
+          annotationCircles.each(function (d, i) {
+            let annotationCircle = d3.select(this);
+            let correspondingRect = d3.select(annotationRects[0][i]);
+            assert.closeTo(TestMethods.numAttr(annotationCircle, "cx"),
+              TestMethods.numAttr(correspondingRect, "x"), window.Pixel_CloseTo_Requirement, `circle ${i} at start of rect`);
+            assert.closeTo(TestMethods.numAttr(annotationCircle, "cy"), 0, window.Pixel_CloseTo_Requirement, `circle ${i} at the top`);
+          });
+
+          let annotationLines = axis.content().selectAll(".annotation-line");
+
+          assert.strictEqual(annotationLines.size(), annotationRects.size(), "same number of annotation lines as rects");
```

Comment:
Probably better to assert that each selection has the same `size()` as the number of ticks.